src/crypto/index.js

```diff
@@ -276,6 +281,60 @@ Crypto.prototype.init = async function() {
     this._checkAndStartKeyBackup();
 };
 
+/**
+ * Bootstrap Secure Secret Storage if needed by creating a default key and signing it with
+ * the cross-signing master key. If everything is already set up, then no
+ * changes are made, so this is safe to run to ensure secret storage is ready
+ * for use.
+ *
+ * @param {function} [opts.doInteractiveAuthFlow] Optional. Function called to
+ * await an interactive auth request.
+ * Args:
+ *     {function} A function that makes the request requiring auth. Receives the
+ *     auth data as an object.
+ */
+Crypto.prototype.bootstrapSecretStorage = async function({
+    doInteractiveAuthFlow,
+} = {}) {
+    logger.log("Bootstrapping Secure Secret Storage");
+
+    // Create cross-signing keys if they don't exist, as we want to sign the SSSS default
+    // key with the cross-signing master key. The cross-signing master keys is also used
+    // to verify the signature on the SSSS default key when adding secrets, so we
+    // effectively need it for both reading and writing secrets.
+    let crossSigningKeysChanged = false;
+    if (!this._crossSigningInfo.getId()) {
+        logger.log("Cross-signing keys not found, creating new keys");
+        await this.resetCrossSigningKeys(
+            CrossSigningLevel.MASTER,
+            { doInteractiveAuthFlow },
+        );
+        crossSigningKeysChanged = true;
+    }
+
+    // Check if Secure Secret Storage has a default key. If we don't have one, create the
+    // default key (which will also be signed by the cross-signing master key).
+    if (!this._secretStorage.hasKey()) {
+        logger.log("Secret storage default key not found, creating new key");
+        const newKeyId = await this.addSecretKey(
```

Comment:
Riot will want to pass a passphrase into this so we'll need to expose that somehow, and can't pass it in each time we start up so I fear we may need to split this function up into "SSSS is set up" and "no? ok, go set it up with this passphrase"

Reply:
Ah right. I guess the main options would be:

* Split it up into a checking only path vs. actual setup as you describe
* Add a UI callback for requesting a passphrase